src/App/Kernel.php

```diff
@@ -400,20 +401,32 @@ public function routes(Application $app)
         }
     }
 
-    public function handleException($e): Response
+    public function handleException(Throwable $e): Response
     {
         /** @var LoggerInterface $logger */
         $logger = $this->get('logger');
-        $logger->error('An unhandled exception was thrown', [
+        if ($e instanceof HttpException) {
+            $logger->error('An http exception was thrown', [
+                'exception' => $e,
+            ]);
+
+            return new JsonResponse(array_filter([
+                'error' => $e->getMessage(),
+                'trace' => $this->app['config']['debug'] ? $e->getTraceAsString() : null,
+            ]), $e->getCode());
+        }
+        $logger->error('An unknown exception was thrown', [
             'exception' => $e,
         ]);
-
         $errorMessage = '
             Internal server error – We are unable to server your request, 
             but it has been logged and we will look into the issue.
         ';
         // This should never be hit, it is a last resort.
-        return new JsonResponse(['error' => trim($errorMessage)], 500);
+        return new JsonResponse(array_filter([
+            'error' => trim($errorMessage),
```

Comment:
I think that also this `error` key, at least when in debug mode, should be the original exception message